boards/FSM/Src/App/App_FlowMeter.c

```diff
@@ -20,11 +31,21 @@ struct FlowMeter *App_FlowMeter_Create(float (*get_flow_rate)(void))
 
     struct FlowMeter *const flow_meter = &flow_meters[alloc_index++];
     flow_meter->get_flow_rate          = get_flow_rate;
+    flow_meter->set_periodic_signal    = set_periodic_signal;
+    flow_meter->can_tx                 = can_tx;
 
     return flow_meter;
 }
 
-float App_FlowMeter_ReadFlowRate(struct FlowMeter *flow_meter)
+void App_FlowMeter_Tick(
+    struct FlowMeter *primary_flow_meter,
+    struct FlowMeter *secondary_flow_meter)
 {
-    return flow_meter->get_flow_rate();
+    const float primary_flow_rate   = primary_flow_meter->get_flow_rate();
```

Comment:
`primary_flow_rate` should be part of `struct FlowMeter` (or rather, as just `flow_rate` because this function should only handle one flow meter at a time. And you should restore `App_FlowMeter_ReadFlowRate` and write it as

```
App_FlowMeter_ReadFlowRate {
  return flow_meter->flow_rate;
}
```

We will need this getter for GTest

Reply:
Added read flow rate